prusti-common/src/vir/ast/common.rs

```diff
@@ -144,6 +149,7 @@ pub enum Type {
     /// TypedRef: the first parameter is the name of the predicate that encodes the type
     TypedRef(String),
     Domain(String),
+    Float,
```

Comment:
Unless I am missing something, this should be:
```suggestion
    Float(Float),
```

Reply:
This and `struct Float` should only exist if Viper has two different float types natively. Does it? Or does it use arbitrary-precision reals? Or both?

Reply:
@Aurel300 As far as I can tell, the BackendType from Viper allows for arbitrary floating-points. `BackendType(s"float${mant}e${exp}", s"(_ FloatingPoint ${exp} ${mant})")` See [src/main/scala/viper/silver/ast/utility/BackendTypeFactories.scala](https://github.com/viperproject/silver/blob/075fd4ba75532b8160429815281fe263b535bac4/src/main/scala/viper/silver/ast/utility/BackendTypeFactories.scala) line 39-73. But Rust only has `f32` and `f64`.

Reply:
I see. In that case `Float(u32, u32)` would map the native types faithfully, but I guess it's unlikely that we would use a non-standard combination of parameters. So I agree with Vytautas' original code suggestion, although maybe I would rename `enum Float` to `enum FloatSize` or similar.